### PR TITLE
Module 4: Add local storage

### PR DESCRIPTION
**Pull request by maxthestranger**
- stored form info in local storage
- populate form fields with the values from local storage

### CODE REVIEW

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @maxthestranger, :wave:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

_See the inline comments below for more required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/main.js [hunk 1]
```diff
@@ -138,14 +138,57 @@ function handleMailValidation(input, msg) {
...
 form.addEventListener('submit', (e) => {
   e.preventDefault();
 
-  handleMailValidation(form.elements.email_address, MAIL_ERROR);
+  const submit = handleMailValidation(form.elements.email_address, MAIL_ERROR);
+
+  if (submit) {
+    storeInfo(form);
+    form.submit();
+  }
```
[the remark this answers is not in the document]
Comment (maxthestranger) [reply js/main.js:171]: Hi @medaminedev66 , so should i capture the inputs values while the user is typing? and store them then

**Review by Hamzaoutdoors — APPROVED** [review]
Hi team, 

## Congratulations! 🎉 Project Approved ✅

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚀

To Highlight:

- [x] The Pull Request has a proper title and description. ✔️
- [x] You have added a descriptive Readme file. Good Job. ✔️
- [x] There is no linter error is present. ✔️
- [x] Project has met all the technical aspects you needed for this project! ✔️ 
- [x] Data is well preserved in the localstorage

_Cheers and Happy coding and Keep rocking!_ 👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/main.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (medaminedev66) [reply js/main.js]: Yes

**Pull request merged**